### PR TITLE
Fix 2255: Rename CppRegion to PyRegion

### LINKED ISSUE #2255: Rename cpp_region to py_algorithms
The current prefix `cpp_` confuses us because it lead us to think it handles regions writen in c++, and not the contrary, i.e. python. To reinforce this, "cppregion" term _already_ is used in nupic.core only to separate code related to cpp regions, not python regions, i.e. more one reason to confusion. Some examples:

https://github.com/numenta/nupic.core/blob/633368e394c3de72e93e401efcd8e91ec30abf53/src/nupic/engine/RegionImplFactory.cpp#L71 and its contrapart to python regions: https://github.com/numenta/nupic.core/blob/633368e394c3de72e93e401efcd8e91ec30abf53/src/nupic/engine/RegionImplFactory.cpp#L60

https://github.com/numenta/nupic.core/blob/fd7757d57fe4ba873156963f961567a5a6f42f43/src/CMakeLists.txt#L427 and its contrapart to python regions: https://github.com/numenta/nupic.core/blob/fd7757d57fe4ba873156963f961567a5a6f42f43/src/CMakeLists.txt#L442

(I prefer use `py_algorithms.so` because is more embracing than `py_region.so`, as in the future this library could include more cortical algorithms code than only python regions code.)

### PR DESCRIPTION
**Pull request by david-ragazzi**
Fixes: #2255 

Depends on merge of https://github.com/numenta/nupic.core/pull/481

### CODE REVIEW

**Comment by david-ragazzi** [comment]
@chetan51 @oxtopus @rhyolight @scottpurdy Please review.

**Comment by rhyolight** [comment]
@david-ragazzi so this depends on https://github.com/numenta/nupic.core/pull/481?

File: setup.py [hunk 1]
```diff
@@ -423,7 +423,7 @@ def getExtensionModules(nupicCoreReleaseDir, platform, bitness, cmdOptions=None)
       ["extensions/cpp_region/PyRegion.cpp",
```
Comment (oxtopus) [inline setup.py]: Need to fixup this and the next line, too, to match `cpp_region` -> `py_algorithms` renaming.

Comment (oxtopus) [inline setup.py]: Also, the travis job should have stopped due to failure at https://travis-ci.org/numenta/nupic/jobs/68411571#L1106, but it continued and failed for a different reason.

Comment (david-ragazzi) [inline setup.py]: Thanks @oxtopus, fixed!

File: setup.py [hunk 2]
```diff
@@ -412,18 +412,18 @@ def getExtensionModules(nupicCoreReleaseDir, platform, bitness, cmdOptions=None)
 
   extensions = []
 
-  libDynamicCppRegion = Extension(
-    "nupic." + getLibPrefix(platform) + "cpp_region",
+  libDynamicPyRegion = Extension(
+    "nupic." + getLibPrefix(platform) + "py_region",
```
Comment (vitaly-krugl) [inline setup.py]: @david-ragazzi: There are additional references to `cpp_region` in setup.py that you may have missed. Please check for other references to `cpp_region` in this and related projects.

**Comment by scottpurdy** [comment]
Now you will need to update `.nupic_modules` based on the latest nupic.core SHA

**Comment by vitaly-krugl** [comment]
@scottpurdy: there is still a problem with this PR - see my preceding comment.

**Comment by david-ragazzi** [comment]
I fixed @vitaly-krugl issue and updated nupic_modules.

**Comment by vitaly-krugl** [comment]
Thank you, @david-ragazzi.

**Comment by david-ragazzi** [comment]
:green_apple:

**Pull request merged**